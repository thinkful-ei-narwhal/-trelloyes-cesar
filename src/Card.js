import React from 'react';

function Card(props) {
  return (<h1>hiii</h1>)
}

export default Card;