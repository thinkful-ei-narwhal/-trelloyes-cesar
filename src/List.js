import React from 'react';


function List(props){
    return(
        <h1></h1>
    );

}

export default List;