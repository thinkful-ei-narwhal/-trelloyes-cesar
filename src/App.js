import React from 'react';
import './App.css';
import List from './List.js';
import STORE from './STORE'


const newRandomCard = () => {
  const id = Math.random().toString(36).substring(2, 4)
    + Math.random().toString(36).substring(2, 4);
  return {
    id,
    title: `Random Card ${id}`,
    content: 'lorem ipsum',
  }
}

function omit(obj, keyToOmit) {
  let {[keyToOmit]: _, ...rest} = obj;
  return rest;
}

class App extends React.Component {
  state = {
    store: STORE
  };

  handleAddItem = (listId) => {
    console.log(listId)
  };

  handleDeleteItem = (cardId) => {
    console.log(cardId)

  };

  render() {
    const { store } = this.state
    return (
      <main className='App'>
        <header className="App-header">
          <h1>Trelloyes!</h1>
        </header>
        <div className="App-list">
          {store.lists.map(list =>
            <List
              key={list.id}
              id={list.id}
              header={list.header}
              cards={list.cardIds.map(id => store.allCards[id])}
              onDeleteItem={this.handleDeleteItem}
              onAddItem={this.handleAddItem}
            />
          )}
        </div>
      </main>
    );
  }
}

export default App;
