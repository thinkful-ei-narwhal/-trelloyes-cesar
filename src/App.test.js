import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import List from './List';
import Card from './Card';
import store from './STORE';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
describe('<Card />', () => {
  it('renders without crashing', () => {
    // Create a DOM element to render the component into
    const div = document.createElement('div');

    // Render the component
    // If something is wrong it will fail here
    ReactDOM.render(<Card
      key={100}
      title={'the card'}
      content={'card string'}
    />, div);

    // Clean up
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    // Render the component, as JSON
    const tree = renderer.create(<Card
      key={100}
      title={'the card'}
      content={'card string'}
    />).toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot();
  });
});