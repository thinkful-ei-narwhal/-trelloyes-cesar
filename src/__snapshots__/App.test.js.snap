// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders this UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    the card
  </h3>
  <p>
    card string
  </p>
</div>
`;
